smithy-model/src/test/resources/software/amazon/smithy/model/errorfiles/loader/json-invalid-list-and-map-members.json

```diff
@@ -0,0 +1,77 @@
+{
+  "smithy": "2.0",
+  "shapes": {
+    "com.test#NoMemberList": {
+      "type": "list"
+    },
+    "com.test#NoMemberMap": {
+      "type": "map"
+    },
+    "com.test#MissingValueMap": {
+      "type": "map",
+      "key": {
+        "target": "smithy.api#String"
+      }
+    },
+    "com.test#MissingKeyMap": {
+      "type": "map",
+      "value": {
+        "target": "smithy.api#String"
+      }
+    },
+    "com.test#WrongMemberList": {
+      "type": "list",
+      "foo": {
```

Comment:
Does this not cause an error because we first check for the required members before checking for extraneous members?

Reply:
No I think we check for extra members first: https://github.com/awslabs/smithy/blob/7342b2c5168dc868c49241e22632a18ca63438bb/smithy-model/src/main/java/software/amazon/smithy/model/loader/AstModelLoader.java#L237, but It just causes a warning: https://github.com/awslabs/smithy/blob/7342b2c5168dc868c49241e22632a18ca63438bb/smithy-model/src/main/java/software/amazon/smithy/model/loader/LoaderUtils.java#L44.

Reply:
Ah I didn't see an error about the extra members in the test. Maybe it's not working as intended.

Reply:
I stepped through the code and I think its this: https://github.com/awslabs/smithy/blob/7342b2c5168dc868c49241e22632a18ca63438bb/smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelAssembler.java#L575
Since there were errors loading, the assembler doesn't return the warnings.

Reply:
Can we make these errors?